src/ucp/rma/amo_send.c

```diff
@@ -15,21 +15,34 @@
 #include <inttypes.h>
 
 
-#define UCP_AMO_CHECK_PARAM(_remote_addr, _size, _opcode, _last_opcode, \
-                            _err_status) \
+#define UCP_AMO_CHECK_PARAM(_context, _remote_addr, _size, _opcode, \
+                            _last_opcode, _status, _action) \
     { \
+        UCP_CONTEXT_CHECK_FEATURE_FLAGS((_context), ((_size) == 4) ? \
+                                        UCP_FEATURE_AMO32 : UCP_FEATURE_AMO64, \
+                                        (_status), _action); \
```

Comment:
i don't understand why need both status and _action here.. can just pass "return INVALID_PARAM" as action, no?
or "return _err_status"?

Reply:
then we need second macro for status ptr as it was for RMA/AMO param check